flink-runtime/src/main/java/org/apache/flink/runtime/webmonitor/threadinfo/JobVertexThreadInfoTracker.java

```diff
@@ -204,31 +206,33 @@ private void triggerThreadInfoSampleInternal(
         }
     }
 
-    private Map<Set<ExecutionAttemptID>, CompletableFuture<TaskExecutorThreadInfoGateway>>
+    private Map<ImmutableSet<ExecutionAttemptID>, CompletableFuture<TaskExecutorThreadInfoGateway>>
             matchExecutionsWithGateways(
                     AccessExecutionVertex[] executionVertices,
                     ResourceManagerGateway resourceManagerGateway) {
 
         // Group executions by their TaskManagerLocation to be able to issue one sampling
         // request per TaskManager for all relevant tasks at once
-        final Map<TaskManagerLocation, Set<ExecutionAttemptID>> executionsByLocation =
+        final Map<TaskManagerLocation, ImmutableSet<ExecutionAttemptID>> executionsByLocation =
```

Comment:
This is not needed strictly speaking but it may ease things, so I'm fine either way. Just wanted to make it explicit. Certainly better than using `? extends Set` here.

Reply:
Yes, I thought it is better to keep it explicit because the value later becomes the key, and we want to make sure we use immutable sets as keys.